src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/FileNameIrregularitiesException.java

```diff
@@ -5,7 +5,7 @@
 
 import java.util.Collection;
 
-public class FileNameIrregularitiesException extends RuntimeException implements EnvelopeAwareThrowable {
+public class FileNameIrregularitiesException extends RuntimeException implements EventRelatedThrowable {
```

Comment:
But we are aware of the envelope 🤔 . Perhaps we should still mark as failure and let the the retry limit allow re-parsing this envelope next time? Just need to be sure that counter is ++'ed

Reply:
The envelope is not created anymore when the files do not match the metadata. This is essentially treated as you'd treat a metadata invalid case as the envelope would be unusable anyway if created.